File: go/api/server-modules/glidejson/options/json_set_options.go
// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
package options

import (
	"github.com/valkey-io/valkey-glide/go/api/options"
)

// This struct represents the optional arguments for the JSON.SET command.
type JsonSetOptions struct {
	ConditionalSet options.ConditionalSet
}

func NewJsonSetOptionsBuilder() *JsonSetOptions {
	return &JsonSetOptions{}
}

// Sets the value only if the given condition is met (within the key or path).
func (jsonSetOptions *JsonSetOptions) SetConditionalSet(conditionalSet options.ConditionalSet) *JsonSetOptions {
	jsonSetOptions.ConditionalSet = conditionalSet
	return jsonSetOptions
}

// Converts JsonGetOptions into a []string.
func (opts JsonSetOptions) ToArgs() ([]string, error) {
	args := []string{}
	var err error
	if opts.ConditionalSet != "" {
		args = append(args, string(opts.ConditionalSet))
	}
	return args, err
}

File: JNI-java/client/src/main/java/glide/api/models/commands/json/JsonSetOptions.java
/** Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0 */
package glide.api.models.commands.json;

import lombok.experimental.SuperBuilder;

/**
 * Options for JSON Set command.
 * This class provides configuration options for setting JSON values.
 */
@SuperBuilder
public class JsonSetOptions {
    
    /**
     * Set operation modes.
     */
    public enum SetMode {
        /** Only set if the key doesn't exist */
        NX,
        /** Only set if the key exists */
        XX
    }
    
    /**
     * The set mode to use.
     */
    private final SetMode mode;
    
    /**
     * GET the set mode.
     * @return The set mode
     */
    public SetMode getMode() {
        return mode;
    }
}